torch/csrc/xpu/Module.cpp

```diff
@@ -247,14 +250,18 @@ static void registerXpuDeviceProperties(PyObject* module) {
       .def_readonly("has_fp64", &DeviceProp::has_fp64)
       .def_readonly("has_atomic64", &DeviceProp::has_atomic64)
       .def_property_readonly("type", get_device_type)
+      .def_property_readonly("device_arch", get_device_architecture)
       .def(
           "__repr__",
-          [&get_device_type, &gpu_subslice_count](const DeviceProp& prop) {
+          [&get_device_type, &gpu_subslice_count, &get_device_architecture](
+              const DeviceProp& prop) {
             std::ostringstream stream;
             stream << "_XpuDeviceProperties(name='" << prop.name
-                   << "', platform_name='" << prop.platform_name << "', type='"
-                   << get_device_type(prop) << "', driver_version='"
-                   << prop.driver_version << "', total_memory="
+                   << "', platform_name='" << prop.platform_name
+                   << "', device_arch='" << get_device_architecture(prop)
```

Comment:
Please use `prop.architecture directly` and remove the lambda `get_device_architecture`